runtime/src/main/scala/akka/grpc/scaladsl/RequestBuilder.scala

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.grpc.scaladsl
+
+import akka.NotUsed
+import akka.annotation.{ DoNotInherit, InternalApi }
+import akka.grpc.internal.ChannelApiHelpers
+import akka.stream.{ Materializer, OverflowStrategy }
+import akka.stream.scaladsl.{ Sink, Source }
+import io.grpc.stub.{ ClientCalls, StreamObserver }
+import io.grpc.{ CallOptions, Channel, Deadline, MethodDescriptor }
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Request builder for requests giving providing per call metadata capabilities
+ *
+ * Not for user extension
+ */
+@DoNotInherit
+sealed trait RequestBuilder[Req, Res] {
+
+  def addMetadata(key: String, value: String): RequestBuilder[Req, Res]
+  def withDeadline(deadline: FiniteDuration): RequestBuilder[Req, Res]
+  def execute(req: Req): Res
```

Comment:
What about `invoke` instead of `execute`?

Reply:
Me neither. ;-) Lagom uses `invoke` hence my slight preference. 

Btw, that API style means that we always have a lifted API representation instead of a switch giving access to the lifted/advanced one. 
I thought we wanted to provide a non-lift, straight, variation for the most common cases.

Reply:
Yeah. Not sure that it is always available rather than hidden behind a switch is a problem though. The old non-lifted API is still there (and should be the default)

Reply:
Aligning with Lagom seems like sensible motivation for `invoke` :)

Reply:
Ah ok, now I see it. Because in gRPC there is always one param, we can overload the method with a paramless variation that gives access to the lifted version. Neat! Much better than a switch.